src/renderer.js

```diff
@@ -40,14 +40,28 @@ document.addEventListener("DOMContentLoaded", () => {
     nav.addEventListener("new-window", () => {
       ipcRenderer.send("new-window");
     });
-
+    function isValidURL(url) {
+      try {
+        if(url.startsWith("peersky://") ) {
+          return false;
+        }
+        new URL(url);
+        return true;
+      } catch (error) {
+        return false;
+      }
+    }
     nav.addEventListener("toggle-bookmark", async () => {
       const urlInput = nav.querySelector("#url");
       if (!urlInput || !urlInput.value) {
         console.error("URL input is empty, cannot toggle bookmark.");
         return;
       }
       const url = urlInput.value.trim();
+      if(!isValidURL(url)) {
+        console.error("Invalid URL format, cannot toggle bookmark.");
+        return;
+      }
```

Comment:
We don't want to restrict them by bookmarking internal URLs. Instead, we aim to fix the issue where all pages were being bookmarked at startup.
https://github.com/p2plabsxyz/peersky-browser/pull/51#discussion_r2202931710